src/mixins/filing-mixin.ts

```diff
@@ -142,6 +142,7 @@ export default class FilingMixin extends Mixins(DateMixin) {
         correctedFilingId: correctedFiling.header.filingId,
         correctedFilingType: FilingTypes.REGISTRATION,
         correctedFilingDate: correctedFiling.header.date,
+        type: correctedFiling.correction.type,
```

Comment:
You can only de-reference this if the corrected filing has a correction object, which is only true is the corrected filing is a correction, which is only true if we allow corrections to corrections, which we haven't agreed to (yet).

Regardless, I think this is indeed the correct place to set type. And maybe also in buildIaCorrectionFiling() if we allow CLIENT ve STAFF for those. Just keep this in mind for now since we haven't decided this either.

Reply:
Sure.

Reply:
If firm corrections always have a "correction" object then this code is fine and you can resolve this conversation.